rct229/rules/section22/section22rule23.py

```diff
@@ -0,0 +1,83 @@
+from rct229.rule_engine.rule_base import RuleDefinitionBase
+from rct229.rule_engine.rule_list_indexed_base import RuleDefinitionListIndexedBase
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.utils.jsonpath_utils import find_all
+
+APPLICABLE_SYS_TYPES = [
+    "SYS-7",
+    "SYS-8",
+    "SYS-11.1",
+    "SYS-11.2",
+    "SYS-12",
+    "SYS-13",
+    "SYS-7B",
+    "SYS-8B",
+    "SYS-11B",
+    "SYS-12B",
+    "SYS-13B",
+]
+
+
+class Section22Rule23(RuleDefinitionListIndexedBase):
+    """Rule 23 of ASHRAE 90.1-2019 Appendix G Section 22 (Hot water loop)"""
+
+    def __init__(self):
+        super(Section22Rule23, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            each_rule=Section22Rule23.ChillerRule(),
+            index_rmr="baseline",
+            id="22-23",
+            description="Each baseline chiller shall be modeled with separate chilled water pump interlocked to operate with the associated chiller.",
+            rmr_context="ruleset_model_instances/0",
+            list_path="chillers[*]",
+        )
+
+    def is_applicable(self, context, data=None):
+        rmi_b = context.baseline
+        # FIXME: replace with baseline_system_types = get_baseline_system_types(rmi_b) when get_baseline_system_types
+        #  is ready.
+        baseline_system_types = {
+            "SYS-7": ["hvac_sys_7"],
+            "SYS-11-1": ["hvac_sys_11_1"],
+        }
+        # if any system type found in the APPLICABLE_SYS_TYPES then return applicable.
+        return any(
+            [key in APPLICABLE_SYS_TYPES for key in baseline_system_types.keys()]
+        )
+
+    def create_data(self, context, data):
```

Comment:
I am confused by the RDS for this rule.  The Rule Description seems to indicate that a check should be made for each chiller. But the Rule Assertion logic seems to indicate a single check for the entire RMI. @weilixu can you enlighten me?

Reply:
I think the RDS has a falsy logic for this rule definition.
The Standard wording is: `**Each** baseline chiller shall be modeled with a separate chilled water pump interlocked to operate with the associated chiller. `

The check shall happen at the component level. I will confirm with the RDS team

Reply:
> I wanna confirm with the RDS team logic before making further comments. I will get back to you once I have a clearer idea on how this should be drafted.

I'll update this PR based on the discussion results with the RDS team.

Reply:
The discussion suggests that this should be at RMI level. @yunjoonjung-PNNL please update accordingly. Thanks!

Reply:
I just changed the code to the RMI level. Could you verify this one more time?